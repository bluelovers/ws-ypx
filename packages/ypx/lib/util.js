Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.crossSpawnOutput = crossSpawnOutput;
exports.stripAnsi = void 0;

var _core = require("cross-spawn-extra/core");

var _crlfNormalize = require("crlf-normalize");

const stripAnsi = _core.CrossSpawnExtra.stripAnsi;
exports.stripAnsi = stripAnsi;

function crossSpawnOutput(buf, options = {
  clearEol: true
}) {
  let output = '';

  if (!Buffer.isBuffer(buf) && Array.isArray(buf)) {
    output = buf.filter(function (b) {
      return !(!b || !b.length);
    }).map(function (b) {
      return b.toString();
    }).join("\n");
  } else {
    output = (buf || '').toString();
  }

  if (options.stripAnsi) {
    output = stripAnsi(output);
  }

  output = (0, _crlfNormalize.crlf)(output);

  if (options.clearEol || options.clearEol == null) {
    output = output.replace(/\n+$/g, '');
  }

  return output;
}
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInV0aWwudHMiXSwibmFtZXMiOlsic3RyaXBBbnNpIiwiQ3Jvc3NTcGF3bkV4dHJhIiwiY3Jvc3NTcGF3bk91dHB1dCIsImJ1ZiIsIm9wdGlvbnMiLCJjbGVhckVvbCIsIm91dHB1dCIsIkJ1ZmZlciIsImlzQnVmZmVyIiwiQXJyYXkiLCJpc0FycmF5IiwiZmlsdGVyIiwiYiIsImxlbmd0aCIsIm1hcCIsInRvU3RyaW5nIiwiam9pbiIsInJlcGxhY2UiXSwibWFwcGluZ3MiOiI7Ozs7OztBQUdBOztBQUNBOztBQUVPLE1BQU1BLFNBQVMsR0FBR0Msc0JBQWdCRCxTQUFsQzs7O0FBRUEsU0FBU0UsZ0JBQVQsQ0FBMEJDLEdBQTFCLEVBQW9FQyxPQUcxRSxHQUFHO0FBQ0hDLEVBQUFBLFFBQVEsRUFBRTtBQURQLENBSEcsRUFNUDtBQUNDLE1BQUlDLE1BQU0sR0FBRyxFQUFiOztBQUVBLE1BQUksQ0FBQ0MsTUFBTSxDQUFDQyxRQUFQLENBQWdCTCxHQUFoQixDQUFELElBQXlCTSxLQUFLLENBQUNDLE9BQU4sQ0FBY1AsR0FBZCxDQUE3QixFQUNBO0FBQ0NHLElBQUFBLE1BQU0sR0FBR0gsR0FBRyxDQUNWUSxNQURPLENBQ0EsVUFBVUMsQ0FBVixFQUNSO0FBQ0MsYUFBTyxFQUFFLENBQUNBLENBQUQsSUFBTSxDQUFDQSxDQUFDLENBQUNDLE1BQVgsQ0FBUDtBQUNBLEtBSk8sRUFLUEMsR0FMTyxDQUtILFVBQVVGLENBQVYsRUFDTDtBQUNDLGFBQU9BLENBQUMsQ0FBQ0csUUFBRixFQUFQO0FBQ0EsS0FSTyxFQVNQQyxJQVRPLENBU0YsSUFURSxDQUFUO0FBVUEsR0FaRCxNQWNBO0FBQ0NWLElBQUFBLE1BQU0sR0FBRyxDQUFDSCxHQUFHLElBQUksRUFBUixFQUFZWSxRQUFaLEVBQVQ7QUFDQTs7QUFFRCxNQUFJWCxPQUFPLENBQUNKLFNBQVosRUFDQTtBQUNDTSxJQUFBQSxNQUFNLEdBQUdOLFNBQVMsQ0FBQ00sTUFBRCxDQUFsQjtBQUNBOztBQUVEQSxFQUFBQSxNQUFNLEdBQUcseUJBQUtBLE1BQUwsQ0FBVDs7QUFFQSxNQUFJRixPQUFPLENBQUNDLFFBQVIsSUFBb0JELE9BQU8sQ0FBQ0MsUUFBUixJQUFvQixJQUE1QyxFQUNBO0FBQ0NDLElBQUFBLE1BQU0sR0FBR0EsTUFBTSxDQUFDVyxPQUFQLENBQWUsT0FBZixFQUF3QixFQUF4QixDQUFUO0FBQ0E7O0FBRUQsU0FBT1gsTUFBUDtBQUNBIiwic291cmNlc0NvbnRlbnQiOlsiLyoqXG4gKiBDcmVhdGVkIGJ5IHVzZXIgb24gMjAyMC8xLzMwLlxuICovXG5pbXBvcnQgeyBTcGF3bkFTeW5jUmV0dXJucywgU3Bhd25BU3luY1JldHVybnNQcm9taXNlLCBJU3Bhd25BU3luY0Vycm9yLCBTcGF3blN5bmNSZXR1cm5zLCBTcGF3bk9wdGlvbnMsIFNwYXduU3luY09wdGlvbnMsIENyb3NzU3Bhd25FeHRyYSB9IGZyb20gJ2Nyb3NzLXNwYXduLWV4dHJhL2NvcmUnO1xuaW1wb3J0IHsgY3JsZiB9IGZyb20gJ2NybGYtbm9ybWFsaXplJztcblxuZXhwb3J0IGNvbnN0IHN0cmlwQW5zaSA9IENyb3NzU3Bhd25FeHRyYS5zdHJpcEFuc2k7XG5cbmV4cG9ydCBmdW5jdGlvbiBjcm9zc1NwYXduT3V0cHV0KGJ1ZjogU3Bhd25TeW5jUmV0dXJuc1tcIm91dHB1dFwiXSB8IEJ1ZmZlciwgb3B0aW9uczoge1xuXHRjbGVhckVvbD86IGJvb2xlYW4sXG5cdHN0cmlwQW5zaT86IGJvb2xlYW4sXG59ID0ge1xuXHRjbGVhckVvbDogdHJ1ZSxcbn0pOiBzdHJpbmdcbntcblx0bGV0IG91dHB1dCA9ICcnO1xuXG5cdGlmICghQnVmZmVyLmlzQnVmZmVyKGJ1ZikgJiYgQXJyYXkuaXNBcnJheShidWYpKVxuXHR7XG5cdFx0b3V0cHV0ID0gYnVmXG5cdFx0XHQuZmlsdGVyKGZ1bmN0aW9uIChiKVxuXHRcdFx0e1xuXHRcdFx0XHRyZXR1cm4gISghYiB8fCAhYi5sZW5ndGgpXG5cdFx0XHR9KVxuXHRcdFx0Lm1hcChmdW5jdGlvbiAoYilcblx0XHRcdHtcblx0XHRcdFx0cmV0dXJuIGIudG9TdHJpbmcoKTtcblx0XHRcdH0pXG5cdFx0XHQuam9pbihcIlxcblwiKVxuXHR9XG5cdGVsc2Vcblx0e1xuXHRcdG91dHB1dCA9IChidWYgfHwgJycpLnRvU3RyaW5nKCk7XG5cdH1cblxuXHRpZiAob3B0aW9ucy5zdHJpcEFuc2kpXG5cdHtcblx0XHRvdXRwdXQgPSBzdHJpcEFuc2kob3V0cHV0KTtcblx0fVxuXG5cdG91dHB1dCA9IGNybGYob3V0cHV0KTtcblxuXHRpZiAob3B0aW9ucy5jbGVhckVvbCB8fCBvcHRpb25zLmNsZWFyRW9sID09IG51bGwpXG5cdHtcblx0XHRvdXRwdXQgPSBvdXRwdXQucmVwbGFjZSgvXFxuKyQvZywgJycpO1xuXHR9XG5cblx0cmV0dXJuIG91dHB1dDtcbn1cbiJdfQ==