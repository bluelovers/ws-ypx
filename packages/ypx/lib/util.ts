/**
 * Created by user on 2020/1/30.
 */
import { SpawnASyncReturns, SpawnASyncReturnsPromise, ISpawnASyncError, SpawnSyncReturns, SpawnOptions, SpawnSyncOptions, CrossSpawnExtra } from 'cross-spawn-extra/core';
import { crlf } from 'crlf-normalize';

export const stripAnsi = CrossSpawnExtra.stripAnsi;

export function crossSpawnOutput(buf: SpawnSyncReturns["output"] | Buffer, options: {
	clearEol?: boolean,
	stripAnsi?: boolean,
} = {
	clearEol: true,
}): string
{
	let output = '';

	if (!Buffer.isBuffer(buf) && Array.isArray(buf))
	{
		output = buf
			.filter(function (b)
			{
				return !(!b || !b.length)
			})
			.map(function (b)
			{
				return b.toString();
			})
			.join("\n")
	}
	else
	{
		output = (buf || '').toString();
	}

	if (options.stripAnsi)
	{
		output = stripAnsi(output);
	}

	output = crlf(output);

	if (options.clearEol || options.clearEol == null)
	{
		output = output.replace(/\n+$/g, '');
	}

	return output;
}
