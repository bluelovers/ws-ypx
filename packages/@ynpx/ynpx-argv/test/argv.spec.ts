import parseArgv from '../index';
import { Arguments } from 'yargs';
import { inspect } from 'util';

test(`test`, () =>
{
	let inputArgv = [
		'-p',
		'ncu2',
		'-p',
		'ncu',
		'ncu',
		'-u',
	];

	let actual = parseArgv(inputArgv);

	expect(actual).toMatchObject({
		package: ['ncu2', 'ncu'],
		_: ['ncu'],
		'--': ['-u'],
	});

	argvSnap(actual);

});

test(`test2`, () =>
{
	let inputArgv = [
		'-p',
		'ncu2',
		'-p',
		'ncu',
		'ncu',
		'--',
		'-u',
	];

	let actual = parseArgv(inputArgv);

	expect(actual).toMatchObject({
		package: ['ncu2', 'ncu'],
		_: ['ncu'],
		'--': ['-u'],
	});

	argvSnap(actual);

});

describe(`test snap`, () => {

	([
		[
			'-q',
			'--ignore-existing',
			'cowsay@latest',
			'--',
			'test'
		],
		[
			'cowsay',
			'-q',
			'--ignore-existing',
			'--',
			'test'
		],
		[
			'-q',
			'--ignore-existing',
			'cowsay',
			'--',
			'test'
		],
		[
			'speedtest',
			'--',
			'-q',
		],
		[
			'speedtest',
			'-q',
		],
		[
			'-p',
			'xxx',
			'yyy',
			'-p',
		],
		'-p esm -p ts-node -p mocha -- -r esm'.split(' '),
		'-p yo -p generator-eslint-typescript yo -- eslint:plugin'.split(' '),
	] as string[][]).forEach(inputArgv => {

		test(inspect(inputArgv), () => {

			let actual = parseArgv(inputArgv);

			argvSnap(actual);

		})

	})

})

function argvSnap(actual: Arguments)
{
	expect(actual).toMatchSnapshot({
		$0: expect.any(String),
	});
}
