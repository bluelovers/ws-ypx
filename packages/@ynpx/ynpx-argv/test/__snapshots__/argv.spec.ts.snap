// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "-u",
  ],
  "_": Array [
    "ncu",
  ],
  "p": Array [
    "ncu2",
    "ncu",
  ],
  "package": Array [
    "ncu2",
    "ncu",
  ],
}
`;

exports[`test snap [
  '-p',
  'yo',
  '-p',
  'generator-eslint-typescript',
  'yo',
  '--',
  'eslint:plugin'
] 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "eslint:plugin",
  ],
  "_": Array [
    "yo",
  ],
  "p": Array [
    "yo",
    "generator-eslint-typescript",
  ],
  "package": Array [
    "yo",
    "generator-eslint-typescript",
  ],
}
`;

exports[`test snap [
  '-p',  'esm',
  '-p',  'ts-node',
  '-p',  'mocha',
  '--',  '-r',
  'esm'
] 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "-r",
    "esm",
  ],
  "_": Array [],
  "p": Array [
    "esm",
    "ts-node",
    "mocha",
  ],
  "package": Array [
    "esm",
    "ts-node",
    "mocha",
  ],
}
`;

exports[`test snap [ '-p', 'xxx', 'yyy', '-p' ] 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "-p",
  ],
  "_": Array [
    "yyy",
  ],
  "p": "xxx",
  "package": "xxx",
}
`;

exports[`test snap [ '-q', '--ignore-existing', 'cowsay', '--', 'test' ] 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "test",
  ],
  "_": Array [
    "cowsay",
  ],
  "ignore-existing": true,
  "ignoreExisting": true,
  "q": true,
  "quiet": true,
}
`;

exports[`test snap [ '-q', '--ignore-existing', 'cowsay@latest', '--', 'test' ] 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "test",
  ],
  "_": Array [
    "cowsay@latest",
  ],
  "ignore-existing": true,
  "ignoreExisting": true,
  "q": true,
  "quiet": true,
}
`;

exports[`test snap [ 'cowsay', '-q', '--ignore-existing', '--', 'test' ] 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "-q",
    "--ignore-existing",
    "--",
    "test",
  ],
  "_": Array [
    "cowsay",
  ],
}
`;

exports[`test snap [ 'speedtest', '--', '-q' ] 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "-q",
  ],
  "_": Array [
    "speedtest",
  ],
}
`;

exports[`test snap [ 'speedtest', '-q' ] 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "-q",
  ],
  "_": Array [
    "speedtest",
  ],
}
`;

exports[`test2 1`] = `
Object {
  "$0": Any<String>,
  "--": Array [
    "-u",
  ],
  "_": Array [
    "ncu",
  ],
  "p": Array [
    "ncu2",
    "ncu",
  ],
  "package": Array [
    "ncu2",
    "ncu",
  ],
}
`;
